src/adapter/src/subscribe.rs

```diff
@@ -113,7 +219,18 @@ impl ActiveSubscribe {
                                     packer.push(Datum::False);
                                 }
 
-                                packer.push(Datum::Int64(diff));
+                                if matches!(self.output, SubscribeOutput::EnvelopeUpsert { .. }) {
+                                    packer.push(match diff {
+                                        -1 => Datum::String("delete"),
+                                        0 => Datum::String("key violation"),
```

Comment:
If indeed this state is here to stay, can we make it `key_violation`, with an underscore instead. This way there are no spaces and quotation marks around that field. I am 100% positive that one day someone will parse the `SUBSCRIBE` stream in a manner that will break if that space is present.

Reply:
done